azhttpclient/middleware.go

```diff
@@ -1,25 +1,37 @@
 package azhttpclient
 
 import (
+	"errors"
 	"fmt"
 	"net/http"
 
 	"github.com/grafana/grafana-azure-sdk-go/azcredentials"
-	"github.com/grafana/grafana-azure-sdk-go/azsettings"
 	"github.com/grafana/grafana-azure-sdk-go/aztokenprovider"
 	"github.com/grafana/grafana-plugin-sdk-go/backend/httpclient"
 )
 
 const azureMiddlewareName = "AzureAuthentication"
 
-func AzureMiddleware(settings *azsettings.AzureSettings, credentials azcredentials.AzureCredentials, scopes []string) httpclient.Middleware {
-	return httpclient.NamedMiddlewareFunc(azureMiddlewareName, func(opts httpclient.Options, next http.RoundTripper) http.RoundTripper {
-		tokenProvider, err := aztokenprovider.NewAzureAccessTokenProvider(settings, credentials)
+func AzureMiddleware(authOpts *AuthOptions, credentials azcredentials.AzureCredentials) httpclient.Middleware {
```

Comment:
`azhttpclient.AddAzureAuthentication` is just a convenience function which adds `AzureMiddleware` to `httpclient.Options`.
I export them both public just in case, alongside with `azhttpclient.ApplyAzureAuth`

Reply:
let's try to minimize the exported functions since any change to them is in theory a breaking change